credstash.py

```diff
@@ -220,15 +220,48 @@ def putSecret(name, secret, version="", kms_key="alias/credstash",
     put a secret called `name` into the secret-store,
     protected by the key kms_key
     '''
+    session = get_session(**kwargs)
+
+    key, contents, hmac = encrypt(
```

Comment:
The main change in this function is that we call `encrypt` in the middle and use its results rather than performing encryption inline.